deepmd/common.py

```diff
@@ -487,3 +487,75 @@ def get_np_precision(precision: "_PRECISION") -> np.dtype:
         return np.float64
     else:
         raise RuntimeError(f"{precision} is not a valid precision")
+
+
+def cast_tensor(input: tf.Tensor,
+                from_precision: tf.DTypes,
+                to_precision: tf.DTypes) -> tf.Tensor:
+    """Convert a Tensor from a precision to another precision.
+    
+    Parameters
+    ----------
+    input: tf.Tensor
+        input tensor
+    precision : tf.DType
+        Tensor data type that casts to
+    
+    Returns
+    -------
+    tf.Tensor
+        casted Tensor
+    
+    Notes
+    -----
+    If input is not a Tensor or without the specific precision, the method will not
+    cast it.
+    """
+    if tf.is_tensor(input) and input.dtype == from_precision:
+        return tf.cast(input, to_precision)
+    return input
+
+
+def cast_precision(func: Callable) -> Callable:
+    """A decorator that casts and casts back the input
+    and output tensor of a method.
```

Comment:
Could you please write more on the logic behind `cast_precision`? 
- It casts i tensor from `GLOBAL_TF_FLOAT_PRECISION` to precision defined by property `precision`.
- It casts o tensor from `precision` to `GLOBAL_TF_FLOAT_PRECISION`.
- It checks the i/o list and only cast when an i or o is tensor and the tensor matches `GLOBAL_TF_FLOAT_PRECISION` or `precision`, respectively.